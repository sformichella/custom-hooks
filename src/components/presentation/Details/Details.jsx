import React from 'react';
import PropTypes from 'prop-types';

export default function Details({ name, image, status }) {
  return (
    <div data-testid="character">
      <h1>{name}</h1>
      <img src={image}/>
      <p>{status}</p>
    </div>
  );
}

Details.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired
};
