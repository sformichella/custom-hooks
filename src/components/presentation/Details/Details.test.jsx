import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Details from './Details';

describe('Details component', () => {
  afterEach(() => cleanup());
  it('renders Details', () => {
    const { asFragment } = render(<Details
      name="Rick Sanchez"
      image="rick-sanchez.com"
      status="alive"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
