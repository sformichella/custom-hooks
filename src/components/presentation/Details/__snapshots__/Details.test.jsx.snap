// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders Details 1`] = `
<DocumentFragment>
  <div
    data-testid="character"
  >
    <h1>
      Rick Sanchez
    </h1>
    <img
      src="rick-sanchez.com"
    />
    <p>
      alive
    </p>
  </div>
</DocumentFragment>
`;
